### PR TITLE
add header only package mscharconv

### PR DESCRIPTION
**Pull request by jingkaimori**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

**Comment by waruqi** [comment]
> Why a commit appearing on master does not submitted to dev?

?

File: packages/m/mscharconv/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                double pi;
+                auto [p, ec] = msstl::from_chars(pi_str.data(), pi_str.data() + pi_str.length(), pi);
+                if (ec != std::errc{}) {
+                    std::cerr << "Error converting \"" << pi_str << "\" to double\n";
+                    return 1;
+                }
+                std::cout << "Got double " << pi << ". The rest is \"" << p << "\".\n";
+                return 0;
+            }
+        ]]}, {configs = {languages = "cxx17"}, includes = {"msstl/charconv.hpp", "iostream", "string_view"}}))
```
Comment (waruqi) [inline packages/m/mscharconv/xmake.lua:42]: please simplify these tests, you can merge into one snippet

File: packages/m/mscharconv/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                std::cout << "Got string " << pi_str << "\n";
+            }
+            void test2(){
+                std::string_view pi_str = "3.14159 is pi";
+                double pi;
+                auto [p, ec] = msstl::from_chars(pi_str.data(), pi_str.data() + pi_str.length(), pi);
+                if (ec != std::errc{}) {
+                    std::cerr << "Error converting \"" << pi_str << "\" to double\n";
+                }
+                std::cout << "Got double " << pi << ". The rest is \"" << p << "\".\n";
```
Comment (waruqi) [inline packages/m/mscharconv/xmake.lua:34]: 我的意思了。两个 test 太多了，尽量精简，否则影响后期仓库拉取大小。。


一个 test 里面大概测试下几个接口，只要 link 能测到就行了。。没必要写这么多，很冗余

**Comment by jingkaimori** [comment]
https://github.com/xmake-io/xmake-repo/pull/2566/commits/0d03b969058c2216c23098f6cdb5e47a81f3e219
this commit appears on master, but not on dev.

**Comment by waruqi** [comment]
> [0d03b96](https://github.com/xmake-io/xmake-repo/commit/0d03b969058c2216c23098f6cdb5e47a81f3e219) this commit appears on master, but not on dev.

这个提交不对。得看看怎么 revert 下

**Comment by waruqi** [comment]
> [0d03b96](https://github.com/xmake-io/xmake-repo/commit/0d03b969058c2216c23098f6cdb5e47a81f3e219) this commit appears on master, but not on dev.

I have reverted it

**Pull request merged**